Cleanup Windows MediaPlayer code

donny-dont opened the pull request:
#### 097123389da2454304dbbd9e85129eae13ea307c
<pre>
Cleanup Windows MediaPlayer code
<a href="https://bugs.webkit.org/show_bug.cgi?id=247654">https://bugs.webkit.org/show_bug.cgi?id=247654</a>

Reviewed by Fujii Hironori.

The code in `MediaPlayerPrivateFullscreenWindow` wasn&apos;t specific to
`ENABLE(VIDEO)` it was specific to `ENABLE(FULLSCREEN_API)` so modified
it to be more general purpose. Updated references of
`MediaPlayerPrivateFullscreenWindow` to `FullscreenWindow` and
`MediaPlayerPrivateFullscreenClient` to `FullscreenClient` and renamed
the files accordingly.

Fixed the WebCore build of `ENABLE(VIDEO)` set to `OFF`. Modified the
linked libraries to be specific to `ENABLE(VIDEO)` and removed unused
ones. Add guards where appropriate and update style to use
`#pragma once`.

* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp:
* Source/WebCore/platform/graphics/win/FullScreenController.cpp:
* Source/WebCore/platform/graphics/win/FullScreenController.h:
* Source/WebCore/platform/graphics/win/FullScreenControllerClient.h:
* Source/WebCore/platform/graphics/win/FullscreenWindow.cpp: Renamed from Source\WebCore\platform\graphics\win\MediaPlayerPrivateFullscreenWindow.cpp.
* Source/WebCore/platform/graphics/win/FullscreenWindow.h: Renamed from Source\WebCore\platform\graphics\win\MediaPlayerPrivateFullscreenWindow.h.
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.h:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/WebKitLegacy/win/FullscreenVideoController.cpp:
* Source/WebKitLegacy/win/FullscreenVideoController.h:
* Source/WebKit/PlatformWin.cmake:
* Source/cmake/OptionsWinCairo.cmake:

Canonical link: <a href="https://commits.webkit.org/256492@main">https://commits.webkit.org/256492@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0a400d9685711e13164e1fd687fee129b22458b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5201 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28995 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105512 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165836 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99931 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5306 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33964 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88334 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101340 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101614 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3908 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82560 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30952 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87670 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73785 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/86981 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39693 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/82323 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37373 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20539 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28225 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4487 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41885 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43156 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/85001 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43793 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39797 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19191 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
LGTM if EWS is happy.

donny-dont committed:
Cleanup Windows MediaPlayer code
https://bugs.webkit.org/show_bug.cgi?id=247654

Reviewed by Fujii Hironori.

The code in `MediaPlayerPrivateFullscreenWindow` wasn't specific to
`ENABLE(VIDEO)` it was specific to `ENABLE(FULLSCREEN_API)` so modified
it to be more general purpose. Updated references of
`MediaPlayerPrivateFullscreenWindow` to `FullscreenWindow` and
`MediaPlayerPrivateFullscreenClient` to `FullscreenClient` and renamed
the files accordingly.

Fixed the WebCore build of `ENABLE(VIDEO)` set to `OFF`. Modified the
linked libraries to be specific to `ENABLE(VIDEO)` and removed unused
ones. Add guards where appropriate and update style to use
`#pragma once`.

* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/PlatformWinCairo.cmake:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.cpp:
* Source/WebCore/platform/graphics/win/FullScreenController.cpp:
* Source/WebCore/platform/graphics/win/FullScreenController.h:
* Source/WebCore/platform/graphics/win/FullScreenControllerClient.h:
* Source/WebCore/platform/graphics/win/FullscreenWindow.cpp: Renamed from Source\WebCore\platform\graphics\win\MediaPlayerPrivateFullscreenWindow.cpp.
* Source/WebCore/platform/graphics/win/FullscreenWindow.h: Renamed from Source\WebCore\platform\graphics\win\MediaPlayerPrivateFullscreenWindow.h.
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.h:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/WebKitLegacy/win/FullscreenVideoController.cpp:
* Source/WebKitLegacy/win/FullscreenVideoController.h:
* Source/WebKit/PlatformWin.cmake:
* Source/cmake/OptionsWinCairo.cmake:

Canonical link: https://commits.webkit.org/256492@main
